File: app/src/main/java/com/sabanciuniv/todoapp/TodoDetails.java
package com.sabanciuniv.todoapp;

import androidx.annotation.NonNull;
import androidx.appcompat.app.AppCompatActivity;

import android.content.Intent;
import android.os.Bundle;
import android.os.Handler;
import android.os.Message;
import android.util.Log;
import android.view.Menu;
import android.view.MenuItem;
import android.widget.TextView;
import android.widget.Toast;

import com.sabanciuniv.todoapp.databinding.ActivityTodoDetailsBinding;
import com.sabanciuniv.todoapp.model.ToDo;
import com.sabanciuniv.todoapp.model.ToDoRepository;

import java.util.concurrent.ExecutorService;

public class TodoDetails extends AppCompatActivity {
    private ActivityTodoDetailsBinding binding;

    Handler handler = new Handler(new Handler.Callback() {
        @Override
        public boolean handleMessage(@NonNull Message msg) {
            Log.i("DELETE", msg.obj.toString());
            finish();


            return true;
        }
    });


    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        binding = ActivityTodoDetailsBinding.inflate(getLayoutInflater());
        setContentView(binding.getRoot());
        ToDo current = (ToDo) getIntent().getSerializableExtra("todoDetails");


        getSupportActionBar().setTitle("Details of To-Do");
        getSupportActionBar().setDisplayHomeAsUpEnabled(true);

        if(current != null){
            binding.seeDetailsTitle.setText(current.getTitle());
            binding.seeDetailsDesc.setText(current.getToDo());
            binding.seeDetailsDueDate.setText(current.getDueDate());
        }
    }


    @Override
    public boolean onCreateOptionsMenu(Menu menu) {

        getMenuInflater().inflate(R.menu.menu_details,menu);
        return true;
    }

    @Override
    public boolean onOptionsItemSelected(@NonNull MenuItem item) {

        if(item.getItemId()==android.R.id.home){
            finish();
        }
        else if(item.getItemId()==R.id.mnDeleteToDo){
        ExecutorService srv = ((ToDoApplication)getApplication()).srv;
        ToDo current = (ToDo) getIntent().getSerializableExtra("todoDetails");

        ToDoRepository repo = new ToDoRepository();
        repo.deleteToDoById(srv, current.getId(), handler);
        finish();
        }
        return true;
    }
}

File: app/src/main/java/com/sabanciuniv/todoapp/activity/TodoDetails.kt
package com.sabanciuniv.todoapp.activity

import android.os.Bundle
import android.os.Handler
import android.util.Log
import android.view.Menu
import android.view.MenuItem
import androidx.appcompat.app.AppCompatActivity
import com.sabanciuniv.todoapp.R
import com.sabanciuniv.todoapp.ToDoApplication
import com.sabanciuniv.todoapp.databinding.ActivityTodoDetailsBinding
import com.sabanciuniv.todoapp.model.ToDo
import com.sabanciuniv.todoapp.repository.ToDoRepository
import java.time.format.DateTimeFormatter

class TodoDetails : AppCompatActivity() {
    private var binding: ActivityTodoDetailsBinding? = null
    var handler = Handler { msg ->
        Log.i("DELETE", msg.obj.toString())
        finish()
        true
    }
    private val dateFormatter = DateTimeFormatter.ofPattern("yyyy-MM-dd HH:mm")

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        binding = ActivityTodoDetailsBinding.inflate(layoutInflater)
        setContentView(binding!!.root)
        val current = intent.getSerializableExtra("todoDetails") as ToDo?
        supportActionBar!!.title = "Details of To-Do"
        supportActionBar!!.setDisplayHomeAsUpEnabled(true)
        if (current != null) {
            binding!!.seeDetailsTitle.text = current.title
            binding!!.seeDetailsDesc.text = current.toDo
            binding!!.seeDetailsDueDate.text = current.dueDate?.format(dateFormatter)
        }
    }

    override fun onCreateOptionsMenu(menu: Menu): Boolean {
        menuInflater.inflate(R.menu.menu_details, menu)
        return true
    }

    override fun onOptionsItemSelected(item: MenuItem): Boolean {
        if (item.itemId == android.R.id.home) {
            finish()
        } else if (item.itemId == R.id.mnDeleteToDo) {
            val srv = (application as ToDoApplication).srv
            val current = intent.getSerializableExtra("todoDetails") as ToDo?
            val repo = ToDoRepository()
            current!!.id?.let { repo.deleteToDoById(srv, it, handler) }
            finish()
        }
        return true
    }
}